Para um primeiro contato com esses dados vou iniciar o projeto reduzindo o escopo separando os arquivos entre os recomendados e os não recomendados , juntamente com a criação de uma cópia na qual serão feitas as análises .

In [2]:
import pandas as pd

doc_item = "2016.csv/Recomendados/item.csv"
# doc_lici = "2016.csv/Recomendados/licitacao.csv"
# doc_item = "2017.csv/Recomendados/item.csv"
# doc_lici = "2017.csv/Recomendados/licitacao.csv"
# doc_item = "2018.csv/Recomendados/item.csv"
# doc_lici = "2018.csv/Recomendados/licitacao.csv"
# doc_item = "2019.csv/Recomendados/item.csv"
# doc_lici = "2019.csv/Recomendados/licitacao.csv"

path_2_save = f"{doc_item}_simplified.csv"
# path_2_save = f"{doc_lici}_simplified.csv"


df_item = pd.read_csv(doc_item , delimiter=",")[["NR_LICITACAO","CD_TIPO_MODALIDADE","DS_ITEM" , "VL_UNITARIO_HOMOLOGADO" ,"QT_ITENS" ,
    "VL_TOTAL_HOMOLOGADO","CD_ORGAO"]]
df_item["NR_LICITACAO___CD_TIPO_MODALIDADE"] = df_item["NR_LICITACAO"].astype(int).astype(str) + "__" + df_item["CD_TIPO_MODALIDADE"].astype(str)
df_item.drop(['NR_LICITACAO', 'CD_TIPO_MODALIDADE'], axis=1, inplace=True)
cols = df_item.columns.tolist()
cols = cols[-1:] + cols[:-1]
df_item = df_item[cols]


# df_item.to_csv(path_2_save , sep = ",")
df_item.head(15)


c:\Users\limaa\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (19,27,28) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


NR_LICITACAO___CD_TIPO_MODALIDADE  \
0                            15__PRP   
1                            15__PRP   
2                            15__PRP   
3                            15__PRP   
4                            15__PRP   
5                            15__PRP   
6                            17__CNV   
7                            17__CNV   
8                            17__CNV   
9                            17__CNV   
10                           17__CNV   
11                           17__CNV   
12                           17__CNV   
13                           17__CNV   
14                           17__CNV   

                                              DS_ITEM  VL_UNITARIO_HOMOLOGADO  \
0   Pneu 1000/20 liso radial, dianteiro, 16 lonas,...                 1430.00   
1                  Pneu 12.4x24 garra baixa, 10 lonas                 1258.00   
2                        Pneu 1400x24 mínimo 12 lonas                 1998.00   
3   Pneu 17.5x25 12 lonas, capacidade mínima de ca...                 2978.00   
4           Pneu 18.4x30 garra baixa, mínimo 10 lonas                 2500.00   
5    Pneu 225/70 r15, 112/110R ou superior, uso misto                  495.00   
6   Xinafoato de salmeterol 25mcg + propionato de ...                   83.73   
7   Rosuvastatina cálcica 10mg – caixa com 30 comp...                   25.89   
8   Policresuleno 50mg/g; cloridrato de cinchocaín...                   15.29   
9   Policarbofila cálcica 500mg – caixa com 30 com...                   24.90   
10     Oxcarbazepina 600mg – caixa com 30 comprimidos                   39.80   
11       Metotrexate 2,5mg – caixa com 24 comprimidos                   21.48   
12  Mesilato de Doxasozina 2mg – caixa com 30 comp...                   15.41   
13       Lamotrigina 100mg – caixa com 30 comprimidos                   31.79   
14         Glimepirida 2mg – caixa com 30 comprimidos                    9.14   

    QT_ITENS VL_TOTAL_HOMOLOGADO  CD_ORGAO  
0       10.0               14300     77200  
1        4.0                5032     77200  
2        6.0               11988     77200  
3        4.0               11912     77200  
4        4.0               10000     77200  
5        8.0                3960     77200  
6       12.0             1004.76     77200  
7       24.0              621.36     77200  
8        6.0               91.74     77200  
9        6.0               149.4     77200  
10      12.0               477.6     77200  
11       6.0              128.88     77200  
12       6.0               92.46     77200  
13      18.0              572.22     77200  
14      30.0               274.2     77200

- Quais são os principais bens materiais comprados pelos órgãos públicos do Rio Grande do Sul?
    - listar os 10 itens mais comprados e qual sua frequencia
    - listar os 10 itens nos quais mais se gasta em compras e qual sua frequencia
    - fazer um histograma para verfificar 

In [73]:
#OBTER A FREQUENCIA DOS MAIS COMPRADOS :
df_buy_freq = None
df_buy_freq = df_item[["DS_ITEM" , "QT_ITENS"]]
# df_buy_freq.columns = ["DS_ITEM" , "QT_ITENS"]

df_buy_freq = df_buy_freq.groupby(["DS_ITEM"] , as_index =False).sum()#

df_buy_freq.loc[df_buy_freq["DS_ITEM"]=="Esfregão de aço"]
df_buy_freq.sort_values("QT_ITENS", ascending= False).head(12)
# print(df_buy_freq)


DS_ITEM    QT_ITENS
233151                                     OMEPRAZOL 20MG  14487032.0
205717            Lote acima de 50.000m² com Edificação    12000000.0
205716                           Lote acima de 50.000m²    12000000.0
233132                                    OMEPRAZOL 20 MG  10864630.0
233148                          OMEPRAZOL 20 mg - Capsula  10700000.0
173175               HIDROCLOROTIAZIDA 25 mg - Comprimido  10447550.0
233203                             OMEPRAZOL CÁPSULA 20MG  10200000.0
300697                                  SINVASTATINA 20MG   9519100.0
254055  PROMETAZINA 25MG                              ...   9000000.0
6974           ACIDO ACETILSALICILICO 100 mg - Comprimido   8833200.0
234443                                     Omeprazol 20mg   8084628.0
234454                           Omeprazol 20mg, cápsula.   7604938.0

- OS ITENS MAIS COMPRADOS EM 2016 :

|   |	DS_ITEM	| QT_ITENS |
| :---  |   :----:  |   ---:|
|233151	| OMEPRAZOL 20MG	| 14487032.0
|233132	|OMEPRAZOL 20 MG	|10864630.0
|233148	|OMEPRAZOL 20 mg - Capsula	|10700000.0
|173175	|HIDROCLOROTIAZIDA 25 mg - Comprimido	|10447550.0
|233203	|OMEPRAZOL CÁPSULA 20MG	|10200000.0
|300697	|SINVASTATINA 20MG	|9519100.0
|254055	|PROMETAZINA 25MG ...	|9000000.0
|6974	|ACIDO ACETILSALICILICO 100 mg - Comprimido	|8833200.0
|234443	|Omeprazol 20mg	|8084628.0
|234454	|Omeprazol 20mg, cápsula.	|7604938.0

In [75]:
#OBTER OS ITENS COM MAIOR IMPACTO FINANCEIRO :
df_buy_freq = None
df_buy_freq = df_item[["DS_ITEM" , "QT_ITENS" , "VL_TOTAL_HOMOLOGADO" ]]
# df_buy_freq.columns = ["DS_ITEM" , "QT_ITENS"]

df_buy_freq = df_buy_freq.groupby(["DS_ITEM"] , as_index =False).sum()#

# df_buy_freq.loc[df_buy_freq["DS_ITEM"]=="Esfregão de aço"]
df_buy_freq.sort_values("QT_ITENS", ascending= False).head(12)
# print(df_buy_freq)


DS_ITEM    QT_ITENS
233151                                     OMEPRAZOL 20MG  14487032.0
205717            Lote acima de 50.000m² com Edificação    12000000.0
205716                           Lote acima de 50.000m²    12000000.0
233132                                    OMEPRAZOL 20 MG  10864630.0
233148                          OMEPRAZOL 20 mg - Capsula  10700000.0
173175               HIDROCLOROTIAZIDA 25 mg - Comprimido  10447550.0
233203                             OMEPRAZOL CÁPSULA 20MG  10200000.0
300697                                  SINVASTATINA 20MG   9519100.0
254055  PROMETAZINA 25MG                              ...   9000000.0
6974           ACIDO ACETILSALICILICO 100 mg - Comprimido   8833200.0
234443                                     Omeprazol 20mg   8084628.0
234454                           Omeprazol 20mg, cápsula.   7604938.0